scripts/local_k8s/unittests/test_cluster.py

```
@@ -0,0 +1,136 @@
+        mocker: pytest_mock.MockerFixture,
+        preview_config: PreviewConfig,
+    ) -> None:
+        """Verify kind teardown skips deletion when the cluster is absent."""
+        config = replace(preview_config, k8s_provider="kind")
+        commands: list[tuple[str, list[str]]] = []
+
+        def record_run(command: str, args: list[str], **_: object) -> CommandResult:
+            commands.append((command, args))
+            return CommandResult(stdout="other\n", stderr="")
+
+        mocker.patch("local_k8s.cluster.require_tools", lambda _: None)
```

Ruff (0.15.21)

[warning] Use `return_value=` instead of patching with `lambda`

(PT008)